### ISSUE TITLE
Cut Down Request Object

### ISSUE THREAD

**Issue by notowen333** [privacy-tech-lab/privacy-pioneer#314]
While working on #312 I realized that we are holding onto (and scanning through) a lot of unnecessary request data. 

I want to cut down the Request Object to only include what we will analyze and remove redundant information.

## What to Keep
1) Request Body
2) Response Data
3) Root/Req Url
4) Url Classifications
5) Type

### Note

For searching, we will only look at items 1, 2, and 3 instead of calling `JSON.stringify` on the whole thing.

## What I Think We Can Remove 

- request headers/responses
     - **Justification**: I don't think we do anthing with this
- The full `details` trait (this is the full object that we only take a few parts of)
     - **Justification**:  The parts we do use for this trait will be specifically drawn out

### Example `details` object:

Including so there is an idea of why to cut this out.

cookieStoreId: "firefox-default"
​
documentUrl: "https://www.nytimes.com/"
​
frameAncestors: Array []
​
frameId: 0
​
incognito: false
​
ip: null
​
method: "GET"
​
originUrl: "https://www.nytimes.com/"
​
parentFrameId: -1
​
proxyInfo: null
​
requestBody: null
​
requestId: "7555"
​
requestSize: 0
​
responseSize: 0
​
tabId: 5
​
thirdParty: true
​
timeStamp: 1629920984771
​
type: "xmlhttprequest"
​
url: "https://c.amazon-adsystem.com/e/dtb/bid?src=3030&u=https%3A%…%22home%22%7D&cfgv=1&gdprl=%7B%22status%22%3A%22no-cmp%22%7D"
​
urlClassification: Object { firstParty: [], thirdParty: (3) […] }

**Comment by SebastianZimmeck** [comment]
On the cookies, I just came across a website, https://www.fastcompany.com/, that stored a cookie named `geodata` with my precise lat/lon coordinates and my ZIP code as the value:

> San Francisco:United States:94110:142.254.99.175

Ideally, we would not lose this type of information. @notowen333, can you check whether that would be available under your current cut down request object?

**Comment by notowen333** [comment]
I checked the https://www.entrepreneur.com/ and this is currently caught in the `main` branch.

With the changes in the `optimize-analysis` branch, these values are not found. As an aside, the Nike case also alerted me that I recreated the video bug that we had way back in the winter in my new branch... Will look for a fix. 

My prefered solution for the cookie issue would be to use the [cookies API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/getAll) to scan through a page's cookies for watchlist items/coordinates. We could do this scan on the [page loaded event](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event). I think this is lower cost than adding headers back. Also, this would allow us to add a `cookie` label if we find information there.
